pkg/libvmi/cloudinit.go

```diff
@@ -57,6 +59,8 @@ func WithCloudInitNoCloudNetworkData(data string) Option {
 
 		volume := getVolume(vmi, cloudInitDiskName)
 		volume.CloudInitNoCloud.NetworkData = data
+		volume.CloudInitNoCloud.NetworkDataBase64 = ""
+		volume.CloudInitNoCloud.NetworkDataSecretRef = nil
```

Comment:
I think that this line should be added elsewhere, too. Eg https://github.com/kubevirt/kubevirt/pull/11310/commits/7b2b782f4ed7b5785bdbab3fe8734638e4611aa7#diff-7cd66099e0fe3bc5ccbe619f796d159e11384d71d06147bae0a6ff7c829284f6R51

Reply:
The referenced function, 'WithCloudInitNoCloudEncodedUserData', configure user-data, it should not override network-data configuration.